keras_cv/datasets/waymo/transformer.py

```diff
@@ -0,0 +1,610 @@
+) -> tf.Tensor:
+    """Extracts point labels.
+
+    Args:
+      box: [M, 7] box tensor.
+      box_class: [M] class of each box.
+      point_smooth_xyz: [N, 3] points.
+
+    Returns:
+      point_box_class: [N] box class of each point.
+    """
+    n = point_smooth_xyz.shape[0]
+    m = box.shape[0]
+    if m == 0:
+        return tf.zeros([n], dtype=tf.int32)
+
+    # [N, M]
+    point_in_box = box_utils.is_within_box_3d(point_smooth_xyz, box)
+    # [N]
+    point_in_any_box = tf.math.reduce_any(point_in_box, axis=-1)
+    # [N]
+    point_box_idx = tf.math.argmax(point_in_box, axis=-1, output_type=tf.int32)
+    # [N]
+    point_box_class = tf.where(point_in_any_box, tf.gather(box_class, point_box_idx), 0)
+
+    return point_box_class
+
+
+def _get_point_label(
+    frame: dataset_pb2.Frame, point_smooth_xyz: tf.Tensor
+) -> struct.LabelTensors:
+    """Extracts labels.
+
+    Args:
+      frame: an open dataset frame.
+      point_smooth_xyz: [N, 3] tensor representing smooth point xyz.
+
+    Returns:
+      Label tensors.
+    """
+    (
+        box_3d,
+        box_meta,
+        box_class,
+        box_id,
+        box_density,
+        box_detection_difficulty,
+    ) = _get_point_label_box(frame)
+    point_box_class = _get_label_point(box_3d, box_class, point_smooth_xyz)
+    box_mask = tf.math.greater(box_class, 0)
+    return struct.LabelTensors(
+        label_box=box_3d,
+        label_box_id=box_id,
+        label_box_meta=box_meta,
+        label_box_class=box_class,
+        label_box_density=box_density,
+        label_box_detection_difficulty=box_detection_difficulty,
+        label_box_mask=box_mask,
+        label_point_class=point_box_class,
+    )
+
+
+def point_vehicle_to_smooth(
+    point_vehicle_xyz: tf.Tensor, sdc_pose: tf.Tensor
+) -> tf.Tensor:
+    """Transforms points from vehicle to smooth frame.
+
+    Args:
+      point_vehicle_xyz: [..., N, 3] vehicle xyz.
+      sdc_pose: [..., 4, 4] the SDC pose.
+
+    Returns:
+      The points in smooth frame.
+    """
+    rot = sdc_pose[..., 0:3, 0:3]
+    loc = sdc_pose[..., 0:3, 3]
+    return (
+        tf.linalg.matmul(point_vehicle_xyz, rot, transpose_b=True)
+        + loc[..., tf.newaxis, :]
+    )
+
+
+def box_3d_vehicle_to_smooth(box_3d: tf.Tensor, sdc_pose: tf.Tensor) -> tf.Tensor:
+    """Transforms 3D boxes from vehicle to smooth frame.
+
+    Args:
+      box_3d: [..., N, 7] 3d boxes in vehicle frame.
+      sdc_pose: [..., 4, 4] the SDC pose.
+
+    Returns:
+      The boxes in smooth frame.
+    """
+    center = box_3d[..., 0:3]
+    dim = box_3d[..., 3:6]
+    heading = box_3d[..., 6]
+
+    new_center = point_vehicle_to_smooth(center, sdc_pose)
+    new_heading = (
+        heading + tf.atan2(sdc_pose[..., 1, 0], sdc_pose[..., 0, 0])[..., tf.newaxis]
+    )
+
+    return tf.concat([new_center, dim, new_heading[..., tf.newaxis]], axis=-1)
+
+
+def build_tensors_from_wod_frame(frame: dataset_pb2.Frame) -> Dict[str, tf.Tensor]:
```

Comment:
let's export this method by adding

`from keras_cv.datasets.waymo.transformer import build_tensors_from_wod_frame`  to `keras_cv/datasets/waymo/__init__.py`

and then make this the default transformer in `load.py` (then we can also make `WOD_FRAME_OUTPUT_SIGNATURE` the default `output_signature`)

Reply:
Done